docs/usage/configuration-options.md

```diff
@@ -594,11 +594,37 @@ Renovate supports two options:
 - `none`: No release filtering (all releases allowed)
 - `strict`: If the release's constraints match the package file constraints, then it's included
 
-We are working on adding more advanced filtering options.
+More advanced filtering options may come in future.
 
-Note: There must be a `constraints` object in your Renovate config for this to work.
+Note: There must be a `constraints` object in your Renovate config, or constraints detected from package files, for this to work.
 This feature is limited to `packagist`, `npm`, and `pypi` datasources.
 
+<!-- prettier-ignore -->
+!!! warning
+    Enabling this feature may result in many package updates being filtered out silently.
+    See below for a description of how it works.
+
+When `constraintsFiltering=strict`, the following logic applies:
+
+- Are there `constraints` for this repository, either detected from source or from config?
+- Does this package's release declare constraints of its own (e.g. `engines` in Node.js)?
+- If so, filter out this release unless the repository constraint is a _subset_ of the release constraint
+
+Here are some examples:
+
+| project                  | dependency   | result   |
```

Comment:
```suggestion
| Project                  | Dependency   | Result   |
```

I like the table! 🥳 Let's capitalize the headers though.

What do you mean with `Project` and `Dependency`? Do you mean something like `Detected constraint` and `Dependency constraint`?

After fixing the table headers you should run `yarn prettier-fix` to format the table. 😉